src/utils/createShopifyUrl.ts

```diff
@@ -8,5 +8,10 @@ export function createShopifyUrl(url: string) {
 
 // Resolves the root URL of the Shopify store and excludes any query parameters.
 function resolveRootUrl() {
+  // Specific clause to handle design mode (Theme Editor mode) in Shopify admin
+  // Shopify adds oseid parameter on href/toString invocation on URL interface
+  if (window.Shopify?.designMode) {
+    return `${window.location.origin}${window.Shopify?.routes?.root}`
```

Comment:
is this branch to preserve parameters?

Reply:
No. This branch is to avoid the dynamic parameter added by Shopify everytime we use `href` or `toString`.
For example, consider root is `/` and origin is `https://example.com`, now if we do the following
```js
new URL(window.Shopify?.routes?.root ?? "/", window.location.origin).href
```
The result would be something like `https://example.com/?oseid=X23cooaDe3232`. Similar result when using `toString`